inference_lib/src/aqlm/inference.py

```diff
@@ -0,0 +1,65 @@
+""" Core mathematics for Additive Quantization (AQ): initialization, reconstruction and beam search"""
+import torch
+import torch.nn as nn
+from aqlm.inference_kernels import forward_pass_quantized_linear
+from aqlm.utils import get_int_dtype
+
+
+class QuantizedLinear(nn.Module):
```

Comment:
please create an issue, **for later**, to examine and potentially deduplicate the code here, 

with a full understanding that we're about as likely to fix it as to never open it

To save you time, here's one possible issue text:

```
In the current version, we're reusing some of the code between `src` and `inference_lib/src`. For instance, inference_lib/src/inference.py:QuantizedLinear resembles src/aq.py:QuantizedLinear.

If we have time, it would be nice to selectively merge some of them.
```

Reply:
My idea was that inference code should be completely separated from the quantisation code not to break the latter and because there isn't much overlap anyway. Those two classes serve very different purpose and share surprisingly little code.

Reply:
I agree with @BlackSamorez on this one.